KarhooUISDK/Workers/KarhooLegalNoticeLinkOpener.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  KarhooLegalNoticeLinkOpener.swift
+//  KarhooUISDK
+//
+//  Created by Bartlomiej Sopala on 02/02/2022.
+//  Copyright © 2022 Flit Technologies Ltd. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class KarhooLegalNoticeLinkOpener {
+    let linkParser: LinkParserProtocol
+    private weak var viewControllerToPresentFrom: UIViewController?
+    
+    init(linkParser: LinkParserProtocol = LinkParser(), viewControllerToPresentFrom: UIViewController?) {
+        self.linkParser = linkParser
+        self.viewControllerToPresentFrom = viewControllerToPresentFrom
+    }
+    
+    func openLink(link: String) {
```

Comment:
Maybe add @discarableResult which is returning true if the link is about to be opened.